### ISSUE TITLE
Extension host terminated unexpectedly

### ISSUE THREAD

**Issue by Sampath-Lokuge** [Microsoft/vscode#25175]
- VSCode Version: 1.11.2
- OS Version: Win 8.1

Below error is appearing every 10 mins.Can you please tell me how to get rid of this? I have installed only 4 extensions as shown below.Thanks.

![Alt Text](http://i.imgur.com/qjbZoCW.png)

![Alt Text](http://i.imgur.com/4Uctjy1.png)

**Comment by Sampath-Lokuge** [comment]
I have disabled all extensions one by one for the workspace.But still, It has the same issue.In other words, this is not the problem of extensions which I have installed.Any idea?

**Comment by Sampath-Lokuge** [comment]
I have worked so nicely with VS code last few months.This issue came after the latest update which I did. Can you please tell me how to rollback to the previous version until you'll give a solution for this annoying issue? I'm working on an Ionic / Angular project. When I reload the workspace I have to restart the live reload server too.This is a huge pain and time-consuming operation. please tell me how to go back to the previous version? @jrieken

**Comment by jaybeecave** [comment]
Same problem. Latest update afaik

**Comment by joaomoreno** [comment]
Can you reproduce the issue in our [insider release](https://code.visualstudio.com/insiders)?

**Comment by Sampath-Lokuge** [comment]
Sure.I have installed the insider and 4 plugins which I use.I'll let you know the result.Thank you @joaomoreno

**Comment by chenyeah** [comment]
Version 1.12.0 also have this problem

**Comment by joaomoreno** [comment]
Also, check the following:

> Just had an awesome chat with @ddzmitry, and we figured out that his issue is that he had a folder open which wasn't a git repository while at the same time there was a git repository in his home folder. This caused Code to handle that git repository which had many many active changes.
>
> This is definitely a dupe of #23976, I'll try to get a fix out today. The fix will try to catch these huge repository issues before they break the extension host.
>
> The workaround for you guys is, which is the *right* fix:
>
> 1. Open the workspace folder in a shell and type `git rev-parse --show-toplevel`. This will tell you where the root of the git repository you have opened is.
> 2. Evaluate whether this repository should exist or not. In @ddzmitry's case it was his user's home, which made no sense. He fixed it by simply removing that `.git` folder.
>
> Once again, thanks @ddzmitry for your help!

**Comment by chenyeah** [comment]
my workspace folder do not have ```.git``` folder,but have ```.svn```folder

**Comment by joaomoreno** [comment]
@chenyeah You might have a separate issue then. Please create a new issue and provide as much detail as possible.

**Comment by chenyeah** [comment]
![image](https://cloud.githubusercontent.com/assets/22478003/25329091/f3ccc376-290c-11e7-8e21-b5ec2260941c.png)
my ```di5cheng```folder have four subfolder  , ach subfolder has a ```.svn```

**Comment by chenyeah** [comment]
![image](https://cloud.githubusercontent.com/assets/22478003/25329342/effcddb6-290d-11e7-95fc-79c4a7c7da3d.png)

my ```di5cheng```folder have five subfolder , each subfolder has a ```.svn```

**Comment by rose-a** [comment]
I am experiencing this issue too...

On a VMware VM running Win7 x64 the message appears every time I open a directory in a git repo which is located on a network share (-> host filesystem or fileserver). 

If I copy the repo over to the VMs hdd I can open the directory without the error appearing. On my host pc (Win 10 x64) I can also open the directories without any problems (even if they are located on a network share).

I have uninstalled any extensions which changed nothing. I also completely reinstalled vs code using the currently available stable installer...

![grafik](https://cloud.githubusercontent.com/assets/15836594/25372624/6d97d3d8-2996-11e7-9461-62ad4cb958c6.png)

So in my case this seems to be related to opening a git repo on a network share in a virtual machine...

**Comment by joaomoreno** [comment]
@rose-a Can you try out our [insider release](https://code.visualstudio.com/insiders)?

@Sampath-Lokuge Still waiting on your reply.

**Comment by rose-a** [comment]
@joaomoreno Same thing...

![grafik](https://cloud.githubusercontent.com/assets/15836594/25374618/64de5174-299e-11e7-82e0-5e325a9d8ac4.png)

**Comment by Sampath-Lokuge** [comment]
Hi @joaomoreno 

I have tested it more than 24 H and no issues for me. In other words, it is working perfectly without any errors.Thank you so much 👍

**Comment by jack-pallot** [comment]
@rose-a if you add `"git.enabled": false` to your settings file, does the issue go away?

**Comment by rose-a** [comment]
@jack-pallot Yep, that removes the error message. Re-enabling git and reloading the editor brings the error back.

Applies to both the stable and the insider version.

**Comment by fiendish** [comment]
@joaomoreno same problem with the insider release

**Comment by joaomoreno** [comment]
@rose-a @fiendish Open the devtools (<kbd>F1</kbd>, `Developer Tools`). Any errors there?

**Comment by fiendish** [comment]
Thousands and growing in https://ticino.blob.core.windows.net/sourcemaps/6eaebe3b9c70406d67c97779468c324a7a95db0e/core/vs/workbench/electron-browser/vs/workbench/electron-browser/file:/Users/code/vsoagent/_work/2/s/src/vs/base/parts/ipc/node/ipc.cp.ts. 

![screenshot from 2017-04-25 15-53-56](https://cloud.githubusercontent.com/assets/201996/25389280/735a9d7a-29d0-11e7-89b8-9c60e0365858.png)

**Comment by joaomoreno** [comment]
@fiendish Your issue is #7950

**Comment by fiendish** [comment]
Will do. Is there a potential solution coming that doesn't require a system setting?

**Comment by joaomoreno** [comment]
@fiendish  Let's go to #25304

Waiting on reply from @rose-a and @fiendish

**Comment by WayneDoherty** [comment]
I have this error too, looking forward to the next release

**Comment by rose-a** [comment]
Sorry for the late reply...

In the devtools I get the following errors (using the stable version)

![2017-05-08 10_58_48-labview 2016 - vmware workstation](https://cloud.githubusercontent.com/assets/15836594/25797449/8f49e0b0-33dd-11e7-8bd4-641438793b71.png)

**Comment by rose-a** [comment]
And this is the result from the insider version after pressing the "Reload this Window" button on the error message:

![2017-05-08 11_05_25-labview 2016 - vmware workstation](https://cloud.githubusercontent.com/assets/15836594/25797649/6a499f98-33de-11e7-8a61-df2fdcbb513a.png)

**Comment by joaomoreno** [comment]
@rose-a Would you have time for a remote screen sharing session?

**Comment by rose-a** [comment]
@joaomoreno Right at the moment I actually have time ;) How would you like to do that?

**Comment by joaomoreno** [comment]
@rose-a Awesome: https://hangouts.google.com/hangouts/_/joaomoreno.com/vscode25175

**Comment by joaomoreno** [comment]
I could easily reproduce in a VMWare VM by calling `fs.watch()`, giving it a shared path from the host. Electron just crashes. Created electron/electron#9416
